### ISSUE TITLE
Pitest creates equivalent mutations

### ISSUE THREAD

**Issue by Vyom-Yadav** [hcoles/pitest#1230]
Mutator: https://pitest.org/quickstart/mutators/#EXPERIMENTAL_MEMBER_VARIABLE

It is also documented that:
> Please Note: This mutator is likely to create equivalent mutations if a member variable is explicitly initialized with the Java default value for the specific type of the member variable as in 
```java
public class EquivalentMutant {
    private int x = 0;
}
```

---

It was recently faced when a `final` field was initialized to `null` in a constructor. Removing the initialization is not possible as it will give a compile-time error.

![Screenshot from 2023-06-12 15-03-23](https://github.com/hcoles/pitest/assets/73882557/9ecd3902-0dfc-4402-a9bc-44cd27618d37)

```java
/**                                                                   
 * The charset used to read the file.                                 
 * {@code null} if the file was reconstructed from a list of lines.   
 */                                                                   
private final Charset charset;                                        
```

---

Though this is already documented, would it be possible to fix this? I.e. to mutate the member variable with the default value only when it has a non-default value?

**Comment by Vyom-Yadav** [comment]
@hcoles ping

**Comment by hcoles** [comment]
Yes, it should now be possible to fix this. A static analysis filtering system was introduced long after this mutator was created that ought to be able to handle this. Similar filters have been created for the default operators, but the non defaults have been overlooked. Thanks for the suggestion.

**Comment by romani** [comment]
@hcoles , is there any plan to fix this issue ?

or may be you just share a bit more links to code on how to do this (based on your comment similar problem was fixed in some other part) and somebody from web can send a fix.
